components/Nav/Nav.test.tsx

```diff
@@ -47,12 +47,25 @@ Object.defineProperty(window, 'matchMedia', {
 });
 
 describe('Nav', () => {
+  beforeAll(() => {
+    const originalCreateElement = document.createElement.bind(document);
```

Comment:
I assume without `bind`, we wouldn't be able to grab document?

Reply:
Can you explain what `bind` does? I'm just curiuos.

Reply:
@choir27 @ryandotfurrer this came about because the test passed, but in the test logs there was a console.error:

![CleanShot 2024-09-18 at 22 02 53](https://github.com/user-attachments/assets/1f02b7a1-9305-4fee-ba13-efde30cdf065)

This mock does the following:

- It calls the original createElement method to create the element.
- If the created element is an anchor tag (<a>), it adds a click event listener that prevents the default action.
- It returns the created (and potentially modified) element.

The purpose of this mock is to prevent anchor tags from triggering actual navigation during tests. By adding the preventDefault() to all anchor tags, the tests can simulate clicks on links without causing unwanted navigation that could interfere with the test environment.

This setup ensures that all calls to document.createElement throughout the tests will use this mocked version, allowing for consistent behavior when testing components that render anchor tags or other HTML elements.